File: Binary_Tree/145-m-postorderTraversal.cpp
// Definition for a binary tree node.
#include <vector> 

using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
};
 
class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> v;
        postTraversal(root, v);
        return v;
    }
    
    void postTraversal(TreeNode* root, vector<int>& v) {
        if(!root)
            return;
        postTraversal(root->left, v);
        postTraversal(root->right, v);
        v.push_back(root->val);
    }
};

File: Binary_Tree/145-m-postorderTraversal.py
# Definition for a binary tree node.
class TreeNode:
     def __init__(self, x):
         self.val = x
         self.left = None
         self.right = None

class Solution:
    def postorderTraversal(self, root: TreeNode):
        res, stack = [], []
        while stack or root:
            if root:
                res += root.val,
                stack += root.left, root.right,
            root = stack.pop()
        return res[::-1]
                
                
        